info-compiler/src/main/java/com/google/sps/webcrawler/NewsContentProcessor.java

```diff
@@ -15,23 +15,236 @@
+        new ArrayList<Vertex<IntValue, StringValue>>(sentences.length);
+    List<Edge<IntValue, DoubleValue>> edges =
+        new ArrayList<Edge<IntValue, DoubleValue>>((int) Math.pow(sentences.length, 2));
+    for (int i = 0; i < sentences.length; i++) {
+      vertices.add(new Vertex(new IntValue(i), new StringValue(sentences[i])));
+      for (int j = i; j < sentences.length; j++) {
+        if (i == j) {
+          edges.add(new Edge(new IntValue(i), new IntValue(j), new DoubleValue(0)));
+          continue;
+        }
+        String[] sentenceA = tokenizeSentence(sentences[i]);
+        String[] sentenceB = tokenizeSentence(sentences[j]);
+        double similarity = computeInterSentenceSimilarity(sentenceA, sentenceB);
+        if (similarity >= SIMILARITY_THRESHOLD) {
+          edges.add(new Edge(new IntValue(i), new IntValue(j), new DoubleValue(similarity)));
+          edges.add(new Edge(new IntValue(j), new IntValue(i), new DoubleValue(similarity)));
+        }
+      }
+    }
+    return Graph.fromCollection(vertices, edges, ExecutionEnvironment.createLocalEnvironment());
+  }
+
+  /**
+   * Finds sentence ranking by applying the PageRank algorithm to {@code similarityGraph} and
+   * sorting all sentences based on descending PageRank scores (high-scored sentence comes first).
+   * If two sentences have the same PageRank score, the sentence that comes early in the original
+   * content comes first.
+   */
+  private static List<PageRank.Result<IntValue>> getRanking(
+      Graph<IntValue, StringValue, DoubleValue> similarityGraph) throws Exception {
+    PageRank pageRank =
+        new PageRank(PAGERANK_DAMPEN_FACTOR, PAGERANK_MAX_ITER, PAGERANK_CONVERGENCE_THRESHOLD);
+    List<PageRank.Result<IntValue>> pageRankResults =
+        pageRank.setIncludeZeroDegreeVertices(PAGERANK_INCLUDE_ZERO_DEGREES_VERTICES)
+            .run(similarityGraph).collect();
+    Collections.sort(pageRankResults, new PAGERANK_SCORE_DESCENDING());
+    return pageRankResults;
+  }
+
+  /**
+   * Extracts the most important {@code SUMMARIZATION_MAX_SENTENCE_NUMBER} sentences for expressing
+   * the overall meaning of {@code sentences}, based on {@code ranking}, and re-arranges the
+   * sentences based on their original order.
+   */
+  private static String extractSentencesBasedOnRanking(List<PageRank.Result<IntValue>> ranking,
+      String[] sentences) {
+    int sentenceNumber = Math.min(SUMMARIZATION_MAX_SENTENCE_NUMBER, sentences.length);
+    List<PageRank.Result<IntValue>> subRanking = ranking.subList(0, sentenceNumber);
+    Collections.sort(subRanking, new SENTENCES_ORIGINAL_ORDER());
+    String summarizedContent = "";
+    for (int i = 0; i < sentenceNumber; i++) {
+      int sentenceIndex = subRanking.get(i).getVertexId0().getValue();
+      summarizedContent += sentences[sentenceIndex];
+      if (i != sentenceNumber - 1) {
+        summarizedContent += " ";
+      }
+    }
+    System.out.println(summarizedContent);
```

Comment:
Was this leftover from debugging?

Reply:
Yes, I overlooked this. I will remove it